src/tabmat/categorical_matrix.py

```diff
@@ -255,7 +283,7 @@ class CategoricalMatrix(MatrixBase):
 
     def __init__(
         self,
-        cat_vec: Union[list, np.ndarray, pd.Categorical],
+        cat_vec,
```

Comment:
mypy might actually be useful here. It is quite good with union types and narrowing them in if-statements these days.

Reply:
`mypy` doesn't play nice with optional imports. I tried conditioning imports on type hinting to no avail.

Reply:
Using [`TYPE_CHECKING`](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) didn't work? I think it is reasonable to assume both Pandas and Polars are installed in the dev environment, isn't it?

Reply:
> Using [TYPE_CHECKING](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) didn't work?

No. :\ I may have done it wrong though.